tests/functional/test_reload.py

```diff
@@ -0,0 +1,148 @@
+# THIS FILE IS PART OF THE ROSE-CYLC PLUGIN FOR THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Functional tests for reinstalling of config files.
+This test does the following:
+
+1. Validates a workflow.
+2. Installs a workflow with some opts set using -O and
+   ROSE_SUITE_OPT_CONF_KEYS.
+3. Re-install workflow.
+4. After modifying the source ``rose-suite.conf``, re-install the flow again.
+
+At each step it checks the contents of
+- ~/cylc-run/temporary-id/rose-suite.conf
+- ~/cylc-run/temporary-id/opt/rose-suite-cylc-install.conf
+"""
+
+import pytest
+from subprocess import run
+
+from pathlib import Path
+
+@pytest.fixture
+def provide_template_vars_workflow(tmp_path):
+    (tmp_path / 'flow.cylc').write_text(
+        '#!jinja2\n'
+        '[scheduling]\n'
+        '    [[graph]]\n'
+        '       R1 = foo\n'
+        '[runtime]\n'
+        '    [[foo]]\n'
+        '        script = cylc message -- {{var}}')
+    (tmp_path / 'rose-suite.conf').write_text(
+        '[template variables]\nvar="rose-suite.conf"')
```

Comment:
N.B. I think it would be easier and cleaner to use a triple quoted string and `textwrap.dedent()`